EXAMPLES.md

```diff
@@ -0,0 +1,202 @@
+        return err
+    }
+
+    // Accumulate here the results or check for a specific client.
+
+    if !clients.HasNext() {
+        break
+    }
+
+    page++
+}
+```
+
+### Checkpoint pagination
+
+Checkpoint pagination can be used when you wish to retrieve more than 1000 results from certain APIs. The APIs that support checkpoint based pagination are:
+
+* `Log.List` (`/api/v2/logs`)
+* `Organization.List` (`/api/v2/organizations`)
+* `Organization.Members` (`/api/v2/organizations/{id}/members`)
+* `Role.Users` (`/api/v2/roles/{id}/users`)
+
+In general to use checkpoint based pagination you should follow a pattern like below:
+
+```go
+var orgList *management.OrganizationList
+for {
+    if orgList == nil {
+        // For the first call, only pass the `take` query parameter, the API will then return a
+        // `Next` value that can be used for future requests,
+        orgList, err = auth0API.Organization.List(
+            management.Take(100),
+        )
+    } else {
+        // Pass the `next` and `take` query parameters now so that we can correctly paginate the
+        // organizations,
+        orgList, err = auth0API.Organization.List(
+            management.From(orgList.Next),
+            management.Take(100),
+        )
+    }
+
+    if err != nil {
+        log.Fatal("err :%w", err)
+    }
+
+    for _, org := range orgList.Organizations {
+        log.Printf("org %s", org.GetID())
+    }
+
+    // The `HasNext` helper func checks whether the API has informed us that there is more data to
+    // retrieve or not.
+    if !orgList.HasNext() {
+        break
+    }
+}
+```
+
+However, for `Log.List`, the `Next` value is not returned via the API but instead is an ID of a log entry and the handling of determining if there are more logs to retrieved must also be done manually.
+
+```go
+var logs []*management.Log
+initialLogId := "LOGID"
+for {
+    // Retrieve 100 logs after the specified log
+    logs, err = auth0API.Log.List(
+        management.From(logFromId),
+        management.Take(100),
+    )
+
+    if err != nil {
+        log.Fatal("err :%w", err)
+    }
+
+    for _, logData := range logs {
+        log.Printf("id %s", logData.GetID())
+        log.Printf("msg %s", logData.GetType())
+
+    }
+
+    // The HasNext helper cannot be used with `Log.List` so instead we check the length of the
+    // returned logs array. When it reached 0 there are no more logs left to process
+    if len(logs) == 0 {
+        break
+    }
+
+    logFromId = logs[len(logs)-1].GetID()
+}
+```
+
+## Providing a custom User struct
+
+The User struct within the SDK only supports the properties supported by Auth0, therefore any extra properties added by an external IDP will not be included within the struct returned from the SDK APIs. In order to expose these custom properties we recommend creating a custom struct and then manually calling the API via the lower level request functionality exposed by the SDK, for example:
```

Comment:
Maybe worth typing out the full IDP acronym, as not everyone will understand straight away what it stands for, wdyt?

Reply:
Yes, we should avoid using domain-specific jargon in developer-facing docs, except for more advance functionality where it's expected that the reader is familiar with it.